data_structures/Heap/Binary_Heap/min_heap_two.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:test/test.dart';
+
+class MinHeap {
+  List<int> heap;
+
+  void buildHeap(List<int> array) {
+    this.heap = _heapify(array);
+  }
+
+  List<int> _heapify(List<int> array) {
+    int firstParent;
+    firstParent = (array.length - 2) ~/ 2;
+    for (int i = firstParent; i >= 0; i--) {
+      _siftDown(i, array.length - 1, array);
+    }
+    return array;
+  }
+
+  int peek() {
+    if (!isEmpty()) {
+      return this.heap[0];
+    }
+    return null;
+  }
+
+  bool isEmpty() {
+    return this.heap.length == 0;
+  }
+
+  void _siftUp(int currentIndex) {
+    int parentIndex;
+    parentIndex = (currentIndex - 1) ~/ 2;
```

Comment:
You could initialize `parentIndex` with this value
```suggestion
    int parentIndex = (currentIndex - 1) ~/ 2;
```